homeassistant/components/sensibo/climate.py

```diff
@@ -138,208 +137,235 @@ async def async_assume_state(service: ServiceCall) -> None:
-            async with async_timeout.timeout(TIMEOUT):
-                data = await self._client.async_get_device(self._id, _FETCH_FIELDS)
-        except (
-            aiohttp.client_exceptions.ClientError,
-            asyncio.TimeoutError,
-            SensiboError,
-        ) as err:
-            if self._failed_update:
-                _LOGGER.warning(
-                    "Failed to update data for device '%s' from Sensibo servers with error %s",
-                    self._attr_name,
-                    err,
-                )
-                self._attr_available = False
-                self.async_write_ha_state()
-                return
-
-            _LOGGER.debug("First failed update data for device '%s'", self._attr_name)
-            self._failed_update = True
-            return
-
-        if self.temperature_unit == self.hass.config.units.temperature_unit:
-            self._attr_target_temperature_step = 1
-        else:
-            self._attr_target_temperature_step = None
-
-        self._failed_update = False
-        self._do_update(data)
+        result = await self._async_set_ac_state_property("on", False)
+        if result:
+            self.coordinator.data[self.unique_id]["on"] = False
+            self.async_write_ha_state()
 
     async def _async_set_ac_state_property(
-        self, name, value, assumed_state=False
-    ) -> None:
+        self, name: str, value: Any, assumed_state: bool = False
+    ) -> bool:
         """Set AC state."""
+        result = {}
         try:
             async with async_timeout.timeout(TIMEOUT):
-                await self._client.async_set_ac_state_property(
-                    self._id, name, value, self._ac_states, assumed_state
+                result = await self._client.async_set_ac_state_property(
+                    self.unique_id,
+                    name,
+                    value,
+                    self.coordinator.data[self.unique_id]["ac_states"],
+                    assumed_state,
                 )
         except (
-            aiohttp.client_exceptions.ClientError,
+            ClientConnectionError,
             asyncio.TimeoutError,
             SensiboError,
         ) as err:
-            self._attr_available = False
-            self.async_write_ha_state()
-            raise Exception(
-                f"Failed to set AC state for device {self._attr_name} to Sensibo servers"
+            raise HomeAssistantError(
+                f"Failed to set AC state for device {self.name} to Sensibo servers: {err}"
             ) from err
+        LOGGER.debug("Result: %s", result)
+        if result["status"] == "Success":
+            return True
+        failure = result["failureReason"]
+        raise HomeAssistantError(
+            f"Could not set state for device {self.name} due to reason {failure}"
+        )
+
+    async def async_assume_state(self, state) -> None:
+        """Sync state with api."""
+        change_needed = state != self.state
```

Comment:
Not sure if there is a simpler way to do it, but I was thinking of something like:
```suggestion
        change_needed = (state == STATE_OFF and self.state != HVAC_MODE_OFF) or (
            state == STATE_ON and self.state == HVAC_MODE_OFF
        )

```

Reply:
```suggestion
        if state == self.state or (state == "on" and self.state != HVAC_MODE_OFF):
```
Is my suggestion at the moment

Reply:
Yes that will work with the suggestion to keep the constant `HAVE_MODE_OFF`